src/main/features/core/windowTitleInfo.js

```diff
@@ -0,0 +1,16 @@
+const windowTitle = WindowManager.get(global.mainWindowID).getTitle();
+let songInfo = {};
+
+if (Settings.get('enableWindowTitleInfo')) {
+  PlaybackAPI.on('change:song', (_songInfo) => {
+    songInfo = _songInfo;
```

Comment:
Saving the song info for state change events, such as play/pause so when the player is paused or stopped the player reverts to the regular title, and when playing again it changes to the Title - Artist. Could change to use PlaybackAPI calls directly.